src/web-push-lib.js

```diff
@@ -220,16 +220,9 @@ WebPushLib.prototype.generateRequestDetails =
     }
 
     const isGCM = subscription.endpoint.indexOf('https://android.googleapis.com/gcm/send') === 0;
+    const isFCM = subscription.endpoint.indexOf('https://fcm.googleapis.com/fcm/send') === 0;
     // VAPID isn't supported by GCM hence the if, else if.
-    if (isGCM) {
```

Comment:
VAPID isn't supported by GCM, so we should keep this if first

Reply:
There is a tradeoff here; if you are using VAPID, then you shouldn't be using GCM endpoint. 

If you are using FCM endpoint to send pushes but without VAPID then this library becomes useless because of the `isGCM` check at the top since FCM is also supported for regular webpushes.

The modification I made allows users to send a regular webpush, w/o VAPID, to both GCM and FCM endpoints which work fine. It does not affect pushed using VAPID.

Reply:
Wouldn't something like:
```JavaScript
if (isGCM) {
  // Use GCM API key
  ...
} else if (currentVapidDetails) {
  // Use VAPID
  ...
} else if (isFCM and currentGCMAPIKey) {
  // Use GCM API key
  ...
}
```
work for you?

Reply:
Yes it would but;
```js
if (isGCM) {
  // The Push logic in this clause is exactly the SAME
} else if (currentVapidDetails) {
} else if (isFCM and currentGCMAPIKey) {
  // with the ONE here
}
```

That is why I'm trying to have;

```js
if (isGCM || isFCM) {
  ...
}
```

Reply:
Yes, I see your point, but I feel more comfortable with this change if we don't change the pre-existing behavior at all, even if the code is almost duplicated.

Reply:
Oh OK, that is your call. Would you like me to commit these changes you wanted ?

Reply:
Yes, thanks!

Reply:
@sarpdoruk will you update the PR?

Reply:
Today, I will.